apps/src/webapp/formStorage.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * CodeOrgApp: Webapp
+ *
+ * Copyright 2014 Code.org
+ *
+ */
+
+'use strict';
+
+/**
+ * Namespace for form storage.
+ */ 
+var FormStorage = module.exports;
+
+
+/**
+ * Creates a new record in the specified table.
+ * @param {string} record.tableName The name of the table to read from.
+ * @param {Object} record Object containing other properties to store
+ *     on the record.
+ * @param {Function} callback Function to call with the resulting record.
+ */
+FormStorage.createRecord = function(record, callback) {
+  var tableName = record.tableName;
+  if (!tableName) {
+    console.log('readRecords: missing required property "tableName"');
+    return;
+  }
+  FormStorage.fetchTableSecret(
+      tableName, 
+      putRecord.bind(this, record, callback));
+};
+
+var putRecord = function(record, callback, tableSecret) {
+  var req = new XMLHttpRequest();
+  req.onreadystatechange = handlePutRecord.bind(req, record, callback, tableSecret);
+  var url = '//' + getFormDataHost() + '/v2/forms/CspTable/' + tableSecret +
+      '/children/CspRecord';
+  delete record.tableName;
```

Comment:
record.tableName is deleted so that it does not show up in the data browser UI. Alternately, I could write this field to storage and simply remove it before displaying it data browser UI.